### PR TITLE
Lasergun radial mode menu

### PR DESCRIPTION
**Pull request by SplinterGP**
## About The Pull Request
Adds a radial menu for changing lasergun modes, autofire did some problem to it so i switched the isnull(location) check for istype screen object like in the autofire refactor PR

it now uses the sprites from the ammohud not these ones, check at diffbot:
image

## Why It's Good For The Game
lasergun mode changing is awfull, takes long, and you need to sweep through all the modes, this makes so you can choose from a simple radial menu after pressing space.

## Changelog
:cl:
qol: lasergun mode changing now uses a radial menu.
balance: delay for switching modes on the lasergun removed.
/:cl:

### CODE REVIEW

**Comment by ilikesaying** [comment]
maybe change it to the ammo counter sprite for much easier selection?

**Comment by BlueHNT** [comment]
I love it, but I still have no clue what the fuck i'm reading.
On another note, how about using the
C:\Users\vojta\Desktop\VSC Bucket\TerraGov-Marine-Corps\icons\mob\ammoHUD.dmi as the radial options so its identical unless a spriter makes the choices
"laser" "laser_overcharge" "laser_spread" ""laser_heat" "laser_disabler"
image

**Comment by Moggram** [comment]
>
>
>maybe change it to the ammo counter sprite for much easier selection?


> 
> 
> I love it, but I still have no clue what the fuck i'm reading.
> On another note, how about using the
> TerraGov-Marine-Corps\icons\mob\ammoHUD.dmi as the radial options so its identical unless a spriter makes the choices
> "laser" "laser_overcharge" "laser_spread" ""laser_heat" "laser_disabler"
> image


TLDR: wyci


I thought about that to. After I looked at the code, it would be a pain in the ass whit the way he wrote it. maybe he could add 

    switch(choice)
        if available_modes = ("laser")
            get_hud_state_image(/datum/ammo/energy/lasgun)
       ///and so on

But I feel like he would have to redo alot of it to make it work. Also it is a nice small bit of code that is doing the radial_menu. If anything, the gun icons should be changed to be more unique.

**Comment by SplinterGP** [comment]
> 
> 
> I love it, but I still have no clue what the fuck i'm reading.
> On another note, how about using the
> TerraGov-Marine-Corps\icons\mob\ammoHUD.dmi as the radial options so its identical unless a spriter makes the choices
> "laser" "laser_overcharge" "laser_spread" ""laser_heat" "laser_disabler"
> image

these seem okay

File: code/modules/projectiles/guns/energy.dm [hunk 1]
```diff
@@ -495,6 +494,8 @@
 	var/message_to_user = ""
 	var/fire_mode = GUN_FIREMODE_SEMIAUTO
 	var/icon_state = "tx73"
+	var/radial_icon = 'icons/mob/radial.dmi'
+	var/radial_icon_state = "laser"
```
Comment (TiviPlus) [inline code/modules/projectiles/guns/energy.dm:499]: Autodoc

File: code/datums/components/full_auto_fire.dm [hunk 2]
```diff
@@ -142,7 +142,7 @@
 	if(get_dist(source.mob, target) < 2) //Adjacent clicking.
 		return
 
-	if(isnull(location)) //Clicking on a screen object.
+	if(istype(target, /obj/screen)) //Clicking on a screen object.
```
Comment (TiviPlus) [inline code/datums/components/full_auto_fire.dm:145]: why

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/datums/components/full_auto_fire.dm (hunk 2 above)
Comment (SplinterGP) [reply code/datums/components/full_auto_fire.dm:145]: Its what caused the problem with not being able to click radials on autofire, the isnull(location) doenst work and it runtimes, on autofire refactor bravmole does this as well

**Review by TiviPlus — APPROVED** [review]
